strawberryfields/configuration.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Configuration
+=============
+
+**Module name:** :mod:`strawberryfields.configuration`
+
+.. currentmodule:: strawberryfields.configuration
+
+This module contains the :class:`Configuration` class, which is used to
+load, store, save, and modify configuration options for Strawberry Fields.
+
+Behaviour
+---------
+
+On first import, Strawberry Fields attempts to load the configuration file `config.toml`, by
+scanning the following three directories in order of preference:
+
+1. The current directory
+2. The path stored in the environment variable ``SF_CONF``
+3. The default user configuration directory:
+
+   * On Linux: ``~/.config/strawberryfields``
+   * On Windows: ``~C:\Users\USERNAME\AppData\Local\Xanadu\strawberryfields``
+   * On MacOS: ``~/Library/Preferences/strawberryfields``
+
+If no configuration file is found, a warning message will be displayed in the logs,
+and all device parameters will need to be passed as keyword arguments when
+loading the device.
+
+The user can access the initialized configuration via `strawberryfields.config`, view the
+loaded configuration filepath, print the configurations options, access and modify
+them via keys, and save/load new configuration files.
+
+Configuration files
+-------------------
+
+The configuration file `config.toml` uses the `TOML standard <https://github.com/toml-lang/toml>`_,
```

Comment:
I never got to the bottom of single or double underscores for links - once you gave a suggestion to change from single to double. Is the single underscore here ok?

Reply:
Yep.  A double underscore just refers to an anonymous hyperlink reference. ReST keeps reference to all hyperlinks even if you intend for them to be anonymous, which can cause errors if you have the same link repeated in the same document. If the same link is not repeated, it's fine to use a single underscore.